src/grafana_service.ts

```diff
@@ -77,7 +77,7 @@ export class DatasourceStream {
       }
     }
 
-    return new Readable({read});
+    return new ReadableStream({ read });
```

Comment:
No, no no. You need return a stream and then create a link it and feed this stream somewhere else. 
You don't need to feed by chunks, but you can't wait until all data is fetched.